linux/deploy_vm/ubuntu/get_mainline_kernel.yml

```diff
@@ -0,0 +1,45 @@
+# Copyright 2022-2024 VMware, Inc.
+# SPDX-License-Identifier: BSD-2-Clause
+---
+- name: "Set the base url for kernel version check"
+  ansible.builtin.set_fact:
+    mainline_kernel_url: "https://kernel.ubuntu.com/mainline"
+
+- name: "Check the kernel version exists"
+  ansible.builtin.uri:
+    url: "{{ mainline_kernel_url }}/v{{ upgrade_kernel_version }}/amd64"
+    return_content: true
+  register: check_kernel_result
+  ignore_errors: yes
+  when: upgrade_kernel_version is match('[0-9]+.[0-9a-z.-]+')
+
+- name: "Get the latest kernel version"
+  when: >
+    (upgrade_kernel_version | lower == 'latest' or
+     upgrade_kernel_version is not match('[0-9]+.[0-9a-z.-]+') or
+     check_kernel_result.status is undefined or check_kernel_result.status != 200)
+  block:
+    - name: "Get the latest kernel with amd64"
+      ansible.builtin.shell: |
+        versions=`curl -s '{{ mainline_kernel_url }}/?C=N;O=D' | sed 's/>/\\n/g' |grep -Po '(?<=v)[0-9]+.[0-9a-z.-]+'|head -10`
```

Comment:
Please make sure `|` has spaces before and after it. Why did this command not work on my Ubuntu 22.04.4 VM?